### DISCUSSION TITLE
Pinia installation error

### DISCUSSION THREAD

**Discussion by hencework** [nuxt/nuxt#28314] [General]
when i try to install pinia in nuxt project (npm install pinia) then i got error: 
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: nuxt-app@undefined
npm ERR! Found: vue@3.4.34
npm ERR! node_modules/vue
npm ERR!  peer vue@"^3.0.0-0 || ^2.6.0" from vue-demi@0.14.10
npm ERR!  node_modules/@floating-ui/vue/node_modules/vue-demi
npm ERR!   vue-demi@">=0.13.0" from @floating-ui/vue@1.1.2
npm ERR!   node_modules/@floating-ui/vue
npm ERR!    @floating-ui/vue@"^1.1.1" from bootstrap-vue-next@0.24.2
npm ERR!    node_modules/bootstrap-vue-next
npm ERR!     bootstrap-vue-next@"^0.24.2" from the root project
npm ERR!  peer vue@"^3.3.4" from @nuxt/vite-builder@3.12.4
npm ERR!  node_modules/@nuxt/vite-builder
npm ERR!   @nuxt/vite-builder@"3.12.4" from nuxt@3.12.4
npm ERR!   node_modules/nuxt
npm ERR!    nuxt@"^3.12.4" from the root project
npm ERR!  13 more (pinia, vue-demi, @unhead/vue, @vitejs/plugin-vue, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! pinia@"*" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: vue@2.6.14
npm ERR! node_modules/vue
npm ERR!  peer vue@">= 2.5 < 2.7" from @vue/composition-api@1.7.2
npm ERR!  node_modules/@vue/composition-api
npm ERR!   peerOptional @vue/composition-api@"^1.4.0" from pinia@2.1.7
npm ERR!   node_modules/pinia
npm ERR!    pinia@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR

**Comment by sirgeon** [comment]
Try clearing the cache and reinstalling:
```sh
npm cache clean --force
rm -rf node_modules
rm package-lock.json
npm install
```

- Update or Downgrade Dependencies since some version of vue/nuxt might not work with the version that pinia has been made for.
- Try downgrading pinia to match with ur other dependencies
- In worst case scenario do npm install pinia --legacy-peer-deps. That should be heavily avoided tho.

**Reply by hencework** [reply]
Thank you for your time.
